src/envoy/http/authn/authenticator_base.cc

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/http/authn/authenticator_base.h"
+#include "src/envoy/http/authn/filter_context.h"
+#include "src/envoy/http/authn/mtls_authentication.h"
+#include "src/envoy/utils/utils.h"
+
+namespace iaapi = istio::authentication::v1alpha1;
+
+namespace Envoy {
+namespace Http {
+namespace Istio {
+namespace AuthN {
+namespace {
+// Returns true if rule is mathed for peer_id
+bool isRuleMatchedWithPeer(const iaapi::CredentialRule& rule,
+                           const std::string& peer_id) {
+  if (rule.matching_peers_size() == 0) {
+    return true;
+  }
+  for (const auto& allowed_id : rule.matching_peers()) {
+    if (peer_id == allowed_id) {
+      return true;
+    }
+  }
+  return false;
+}
+}  // namespace
+
+AuthenticatorBase::AuthenticatorBase(
+    FilterContext* filter_context,
```

Comment:
use FilterContext&

Reply:
I like to keep pointer as object is mutable (per google style). Plus, StrickMock will convert it to const ref otherwise, make it unusable for test.